<a href="https://colab.research.google.com/github/Yashmitha18/AIML-Project/blob/main/heart_disease.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Features of datasets**

In [ ]:
import pandas as pd
df = pd.read_csv('https://talentcocomedia.s3.amazonaws.com/ml-assets/heart-disease.csv')
df.head()

age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  slope  \
0   63    1   3       145   233    1        0      150      0      2.3      0   
1   37    1   2       130   250    0        1      187      0      3.5      0   
2   41    0   1       130   204    0        0      172      0      1.4      2   
3   56    1   1       120   236    0        1      178      0      0.8      2   
4   57    0   0       120   354    0        1      163      1      0.6      2   

   ca  thal  target  
0   0     1       1  
1   0     2       1  
2   0     2       1  
3   0     2       1  
4   0     2       1

In [ ]:
df.shape #rows, colums

(303, 14)

The data set contains a total of 76 attributes out of which 14 will be used

1. age: age in years
2. sex: sex(1=male;0=female)
3. cp: chest pain type, Value 1: typical angina, Value 2: atypical angina, Value 3: non-anginal pain, Value 4: asymptomatic
4. trestbps: resting blood pressure 
5. chol: serum cholestoral in mg/dl
6. fbs: (fasting blood sugar > 120 mg/dl) (1 = true; 0 = false)
7. restecg: resting electrocardiographic results. Value 0: normal, Value 1: having ST-T wave abnormality (T wave inversions and/or ST elevation or depression of > 0.05 mV), Value 2: showing probable or definite left ventricular hypertrophy by Estes' criteria.
8.thalach: maximum heart rate achieved
9.exang: exercise induced angina (1 = yes; 0 = no)
10.oldpeak = ST depression induced by exercise relative to rest
11.slope: the slope of the peak exercise ST segment. Value 1: upsloping, Value 2: flat, Value 3: downsloping
12.ca: number of major vessels (0-3) colored by flourosopy
13.thal: 3 = normal; 6 = fixed defect; 7 = reversable defect
14.The predicted attribute

In [1]:
import pandas as pd
df=pd.read_csv('https://talentcocomedia.s3.amazonaws.com/ml-assets/heart-disease.csv')
df.head()

age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  slope  \
0   63    1   3       145   233    1        0      150      0      2.3      0   
1   37    1   2       130   250    0        1      187      0      3.5      0   
2   41    0   1       130   204    0        0      172      0      1.4      2   
3   56    1   1       120   236    0        1      178      0      0.8      2   
4   57    0   0       120   354    0        1      163      1      0.6      2   

   ca  thal  target  
0   0     1       1  
1   0     2       1  
2   0     2       1  
3   0     2       1  
4   0     2       1

In [3]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [4]:
df.tail()

age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  \
298   57    0   0       140   241    0        1      123      1      0.2   
299   45    1   3       110   264    0        1      132      0      1.2   
300   68    1   0       144   193    1        1      141      0      3.4   
301   57    1   0       130   131    0        1      115      1      1.2   
302   57    0   1       130   236    0        0      174      0      0.0   

     slope  ca  thal  target  
298      1   0     3       0  
299      1   0     3       0  
300      1   2     3       0  
301      1   1     3       0  
302      1   1     2       0

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 303 entries, 0 to 302
Data columns (total 14 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       303 non-null    int64  
 1   sex       303 non-null    int64  
 2   cp        303 non-null    int64  
 3   trestbps  303 non-null    int64  
 4   chol      303 non-null    int64  
 5   fbs       303 non-null    int64  
 6   restecg   303 non-null    int64  
 7   thalach   303 non-null    int64  
 8   exang     303 non-null    int64  
 9   oldpeak   303 non-null    float64
 10  slope     303 non-null    int64  
 11  ca        303 non-null    int64  
 12  thal      303 non-null    int64  
 13  target    303 non-null    int64  
dtypes: float64(1), int64(13)
memory usage: 33.3 KB


In [6]:
df.isnull().sum()

age         0
sex         0
cp          0
trestbps    0
chol        0
fbs         0
restecg     0
thalach     0
exang       0
oldpeak     0
slope       0
ca          0
thal        0
target      0
dtype: int64

Here none of the attributes are null for any given row and hence we are not going to any drop column.

According to the dataset shown above, all of the attributes play a significant role in identifying heart disease; for this reason, we retain all of the attributes and don't remove any of them.

**Notes**

****

**Continuous Data**

Numerical data that is continuous can have any value as long as it falls within a given range or period. Time, height, weight, temperature, and distance are some examples of continuous variables that are frequently measured on a scale with an unlimited number of possible values and that can be divided into ever-smaller fractions. In a line or curve graph, continuous data can be represented as decimal fractions or irrational numbers.

**Categorical data**

Categorical data is a type of data that represents characteristics or qualities of a particular group or population. Categorical data is typically divided into two types: nominal and ordinal data. Nominal data represents categories that have no inherent order or ranking, while ordinal data represents categories that have a specific order or ranking.

Heatmap, pair plot, and cluster map are all data visualization techniques that are commonly used in data analysis and machine learning tasks. They provide a visual representation of the relationships between variables in a dataset and can help to identify patterns, correlations, and clusters.

**Heatmap**

Heat maps are visual displays of data that depict values in a matrix using color-coded cells. Heat maps are frequently employed to show patterns and trends in huge datasets, and they are especially helpful when the data is too complicated for human analysis. Understanding the significance of the colours and patterns in the cells is necessary for correctly interpreting heat maps.

**Pair Plot**

A pair plot shows the pairwise relationships between variables in a dataset graphically. It is sometimes referred to as a pairs plot or a scatterplot matrix. The pair plot displays histograms of the variables on the diagonal as well as scatterplots between each pair of variables in the dataset.
Particularly when there are numerous variables involved, pair plots are helpful for examining the relationships between variables in a dataset. They can aid in finding outliers or other anomalies in the data, as well as patterns or relationships between different variables.

**Cluster map**

A cluster map is a graphical depiction of the grouping of data based on their similarity or dissimilarity. It is also known as a dendrogram heat map or a hierarchical cluster analysis plot. Finding groups of objects or variables that share similar properties is a typical task in data analysis, especially in the field of bioinformatics. A dendrogram and a heat map are the two main parts of a cluster map. A diagram that resembles a tree called a dendrogram is used to depict the hierarchical relationships between groups of items or variables. The data values for each object or variable are displayed on the heat map, which is typically represented by colour.
